### PR TITLE
Allow to use basic html to format invite messages

### PR DESCRIPTION
**Pull request by skolmer**
Fixes https://github.com/vector-im/element-web/issues/15738
Type: enhancement

---

This PR introduces a new field `io.element.html_reason` that can be set on `m.room.member` state events to get basic html formatting in invite messages. It extends a new feature introduced in https://github.com/matrix-org/matrix-react-sdk/pull/5694

image

```json
{
  "content": {
    "avatar_url": "mxc://matrix.org/test",
    "displayname": "Test",
    "membership": "join",
    "reason": "Daily Meeting, Start 10:00 AM",
    "io.element.html_reason": "<table><tr><td><b>Topic</b></td><td>Daily Meeting</td></tr><tr><td><b>Start</b></td><td>10:00 AM</td></tr></table>"
  },  
  "state_key": "@test:matrix.org",
  "type": "m.room.member"
}
```

Signed-off-by: Steffen Kolmer <steffen.kolmer@nordeck.net>

---
Here's what your changelog entry will look like:

## ✨ Features
 * Allow to use basic html to format invite messages ([\#6703](https://github.com/matrix-org/matrix-react-sdk/pull/6703)). Fixes vector-im/element-web#15738 and vector-im/element-web#15738. Contributed by [skolmer](https://github.com/skolmer).

Preview: https://612fe62fa1edf02da15b34c9--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/rooms/RoomPreviewBar.js [hunk 1]
```diff
@@ -492,9 +492,13 @@ export default class RoomPreviewBar extends React.Component {
...
                 const myUserId = MatrixClientPeg.get().getUserId();
-                const reason = this.props.room.currentState.getMember(myUserId).events.member.event.content.reason;
-                if (reason) {
-                    reasonElement = <InviteReason reason={reason} />;
+                const memberEventContent = this.props.room.currentState.getMember(myUserId).events.member.event.content;
+
+                if (memberEventContent.reason) {
+                    reasonElement = <InviteReason
+                        reason={memberEventContent.reason}
+                        htmlReason={memberEventContent["io.element.html_reason"]}
```
Comment (SimonBrandner) [inline src/components/views/rooms/RoomPreviewBar.js:500]: I think I would feel a bit better if `io.element.html_reason` were to be a `const` somewhere, though I don't know where exactly. Perhaps someone from the team will have an idea

**Review by SimonBrandner — APPROVED** [review]
Thanks for working on this, it looks great! Just a super tiny nitpick

**Review by SimonBrandner — APPROVED** [review]

File: src/components/views/rooms/RoomPreviewBar.js (hunk 1 above)
Comment (skolmer) [reply src/components/views/rooms/RoomPreviewBar.js]: It is a `const` on module level now. If there is a better place to put it please let me know.

File: src/components/views/rooms/RoomPreviewBar.js [hunk 2]
```diff
@@ -492,9 +494,13 @@ export default class RoomPreviewBar extends React.Component {
                 }
 
                 const myUserId = MatrixClientPeg.get().getUserId();
-                const reason = this.props.room.currentState.getMember(myUserId).events.member.event.content.reason;
-                if (reason) {
-                    reasonElement = <InviteReason reason={reason} />;
+                const memberEventContent = this.props.room.currentState.getMember(myUserId).events.member.event.content;
```
Comment (turt2live) [inline src/components/views/rooms/RoomPreviewBar.js:497]:
```suggestion
                const memberEventContent = this.props.room.currentState.getMember(myUserId).events.member.getContent();
```

**Review by turt2live — APPROVED** [review]
otherwise looks good - thanks!

**Pull request merged**